### PR TITLE
telemetry: add stackdriver metrics sink

### PR DESCRIPTION
**Pull request by tam7t**
Adds a way to have vault write telemetry data to stackdriver directly.

This should help users of https://github.com/sethvargo/vault-on-gke and other GCP vault deployments better monitor vault performance without having to use a [collectd sidecar](https://cloud.google.com/monitoring/agent/plugins/statsd).

### CODE REVIEW

**Comment by tam7t** [comment]
Hey @jefferai wondering if you could take a look at this and whether it's something you all would accept, Thanks!

**Comment by malnick** [comment]
@tam7t we are reviewing this today, thanks for your patience and for the contribution!

File: website/source/docs/configuration/telemetry.html.md [hunk 1]
```diff
@@ -157,9 +157,42 @@ bearer_token: your_vault_token_here
...
+*  `stackdriver_project_id` the Google Cloud ProjectID to send telemetry data to.
+*  `stackdriver_location` the GCP or AWS region of the monitored resource.
+*  `stackdriver_namespace` a namespace identifier for the telemetry data.
+
+It is recommended to also enable the option `disable_hostname` to avoid having prefixed
+metrics with hostname. 
+
 ```hcl
 telemetry {
-  prometheus_retention_time = "30s",
```
Comment (kalafut) [inline website/source/docs/configuration/telemetry.html.md]: I think this inadvertently deleted the existing Prometheus example config.

**Comment by kalafut** [comment]
Just the one comment about docs, otherwise LGTM.

File: website/source/docs/configuration/telemetry.html.md (hunk 1 above)
Comment (tam7t) [reply website/source/docs/configuration/telemetry.html.md]: Oh no! Fixed.

**Comment by tam7t** [comment]
Pushed fix, I haven't included `go mod tidy` or `go mod vendor` in the PR. Let me know if I should create commits for those or not, Thanks for the review!

**Comment by kalafut** [comment]
@tam7t `go mod tidy` would be appreciated, but vendoring isn't needed (nor wanted 🙂).

**Comment by tam7t** [comment]
Added the `go mod tidy`

**Pull request merged**